shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureWrapper.java

```diff
@@ -52,7 +52,11 @@ public void release() {
   // Called by native.
   @SuppressWarnings("unused")
   public void attachToGLContext(int texName) {
-    surfaceTexture.attachToGLContext(texName);
+    synchronized (this) {
+      if (!released) {
```

Comment:
Wont this same synchronization be necessary for the other methods here? I spot `detachFromGLContext` and `getTransformMatrix`.

Reply:
Method `getTransformMatrix`  called after `SurfaceTexture.release` should not throw a error like other methods according to https://android.googlesource.com/platform/frameworks/native/+/master/libs/gui/GLConsumer.cpp#713

Method `detachFromGLContext ` is called when `rasterizer->Teardown()` ，At this time, the platform thread is blocked， https://github.com/flutter/engine/blob/master/shell/common/shell.cc#L866

Reply:
In addition, we did not find the crash caused by these two methods.